TileStache/Goodies/VecTiles/transform.py

```diff
@@ -1156,6 +1159,283 @@ def exterior_boundaries(feature_layers, zoom,
+    linestring, or a geometry collection with lines and points in it.
+    So this can be thought of as a "safer" wrapper around Shapely's
+    function.
+    """
+    geom_type = geom.type
+
+    if geom_type == 'GeometryCollection':
+        # collect together everything line-like from the
+        # geometry collection
+        lines = map(_linemerge, geom.geoms)
+
+        # filter out anything that's empty
+        lines = filter(lambda g: not g.is_empty, lines)
+
+        if len(lines) == 0:
+            return MultiLineString([])
+        else:
+            return linemerge(lines)
+
+    elif geom_type == 'LineString':
+        return geom
+
+    elif geom_type == 'MultiLineString':
+        return linemerge(geom)
+
+    else:
+        return MultiLineString([])
+
+
+def admin_boundaries(feature_layers, zoom, base_layer,
+                     start_zoom=0):
+    """
+    Given a layer with admin polygons and maritime boundaries,
+    attempts to output a set of oriented boundaries with properties
+    from both the left and right polygon, and also cut with the
+    maritime information to provide a `maritime_boundary=yes` value
+    where there's overlap between the maritime lines and the
+    polygon boundaries.
+    """
+
+    layer = None
+
+    # don't start processing until the start zoom
+    if zoom < start_zoom:
+        return layer
+
+    layer = _find_layer(feature_layers, base_layer)
+    if layer is None:
+        return None
+
+    # layer will have polygonal features for the admin
+    # polygons and also linear features for the maritime
+    # boundaries. further, we want to group the admin
+    # polygons by their kind, as this will reduce the
+    # working set.
+    admin_features = defaultdict(list)
+    maritime_features = list()
+    new_features = list()
+
+    for shape, props, fid in layer['features']:
+        dims = _geom_dimensions(shape)
+        kind = props.get('kind')
+        maritime_boundary = props.get('maritime_boundary')
+
+        # note: dims=4 means polygon, as it's a bit set of
+        # 1 << num_dimensions. likewise, dims=2 means a
+        # linestring. the reason to use this rather than
+        # compare the string of types is to catch the
+        # "multi-" types as well.
+        if dims == 4 and kind is not None:
+            admin_features[kind].append((shape, props, fid))
+
+        elif dims == 4 and maritime_boundary == 'yes':
+            maritime_features.append((shape, {'maritime_boundary':'no'}, 0))
+
+    # there are separate polygons for each admin level, and
+    # we only want to intersect like with like because it
+    # makes more sense to have Country-Country and
+    # State-State boundaries (and labels) rather than the
+    # (combinatoric) set of all different levels.
+    for kind, features in admin_features.iteritems():
+        num_features = len(features)
+        envelopes = map(lambda g: g[0].envelope, features)
```

Comment:
Yup, that's probably more Pythonic (even if it still tastes slightly functional :wink:). Fixed in cd5643e.